### PR TITLE
telescope: update to 0.9

### PR DESCRIPTION
**Pull request by aeiouaeiouaeiouaeiouaeiouaeiou**
* fix build on macOS < 10.13

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 10K549 x86_64
Xcode 3.2.6 10M2518

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: net/telescope/Portfile [hunk 1]
```diff
@@ -2,26 +2,30 @@
...
+                    sha256  b556d4d741e0ad6fa7bf31abba2c5feb135819d141ee851c9a4ef13ceb798c8c \
+                    size    2374080
 
 depends_build       port:autoconf \
                     port:bison \
                     port:pkgconfig
 
 depends_lib         path:lib/libtls.dylib:libretls \
-                    port:libevent \
+                    port:libgrapheme \
```
Comment (sikmir) [inline net/telescope/Portfile:28]: Just to be sure:
```
post-patch {
    # Remove bundled libraries
    file delete -force ${worksrcpath}/libgrapheme
}
```

**Comment by aeiouaeiouaeiouaeiouaeiouaeiou** [comment]
@sikmir I'll look at it later today, turning the PR into a draft for now.

**Comment by omar-polo** [comment]
@sikmir the new code to generate client certs needs to use functions from libcrypto and libssl, so 0.9 needs to link to them as well now.  It should work however with both LibreSSL and OpenSSL implementations (maybe with a few warnings about deprecated functions), as long as you stick with just one of the two (i.e. libretls + OpenSSL or LibreSSL)

**Comment by sikmir** [comment]
> the new code to generate client certs needs to use functions from libcrypto and libssl, so 0.9 needs to link to them as well now.

Oh, I see.

> It should work however with both LibreSSL and OpenSSL implementations (maybe with a few warnings about deprecated functions), as long as you stick with just one of the two (i.e. libretls + OpenSSL or LibreSSL)

Well, libretls + OpenSSL is our case.

**Comment by aeiouaeiouaeiouaeiouaeiouaeiou** [comment]
@sikmir @omar-polo I rebuilt the package on 10.6 machine with OpenSSL installed and it built successfully, but now had a strange effect when running. There's no TUI and only the letter 's' appears (same on iTerm 2):

image

Nothing suspicious in otool:
```
otool -L $(which telescope)
/opt/local/bin/telescope:
        /opt/local/lib/libMacportsLegacySupport.dylib (compatibility version 1.0.0, current version 1.2.1)
        /opt/local/libexec/openssl3/lib/libssl.3.dylib (compatibility version 3.0.0, current version 3.0.0)
        /opt/local/libexec/openssl3/lib/libcrypto.3.dylib (compatibility version 3.0.0, current version 3.0.0)
        /opt/local/lib/libtls.28.dylib (compatibility version 29.0.0, current version 29.0.0)
        /opt/local/lib/libncurses.6.dylib (compatibility version 6.0.0, current version 6.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 125.2.11)
```

This is starting to remind me of the story in #19047, but `legacysupport` uses the [correct](https://github.com/macports/macports-legacy-support/commit/d24f1d50c27f86fa05aa0fac209b6468b07f4940) `open_memstream` implementation.

**Comment by omar-polo** [comment]
don't have a mac to verify but 

```
        /opt/local/libexec/openssl3/lib/libssl.3.dylib (compatibility version 3.0.0, current version 3.0.0)
        /opt/local/libexec/openssl3/lib/libcrypto.3.dylib (compatibility version 3.0.0, current version 3.0.0)
        /opt/local/lib/libtls.28.dylib (compatibility version 29.0.0, current version 29.0.0)
```

isn't this still mixing libressl (libtls) with openssl?  I would expect to see `lib*re*tls.dylib` here.

(The names are utterly confusing and I'm afraid I can't do much for this.  I can add a check in the configure to catch these mixing of incompatible libraries however.)

> This is starting to remind me of the story in https://github.com/macports/macports-ports/pull/19047, but legacysupport uses the [correct](https://github.com/macports/macports-legacy-support/commit/d24f1d50c27f86fa05aa0fac209b6468b07f4940) open_memstream implementation.

just for curiosity, is open_memstream tricky to handle on macos?  I could eventually stop using it upstream if it helps.

**Comment by aeiouaeiouaeiouaeiouaeiouaeiou** [comment]
>edit: sorry for the noise, I got confused with the naming of the libraries myself, apologise.

It doesn't help though, still does the same library linking and crash.

CC: @barracuda156 for test

**Comment by reneeotten** [comment]
any progress on this? If not, please close until someone had the time/interest to sort this out.

**Comment by barracuda156** [comment]
@sikmir Why do we need to mix the two crypto libs though?

**Comment by barracuda156** [comment]
Does 0.8.1 work on older macOS by the way? If yes, let’s just offer a fallback and add a link to this discussion with a FIXME in the portfile.

**Comment by sikmir** [comment]
0.10 fixes the issue, it works for me.

**Comment by barracuda156** [comment]
> 0.10 fixes the issue, it works for me.

Could you make a PR?

**Comment by sikmir** [comment]
> > 0.10 fixes the issue, it works for me.
> 
> Could you make a PR?

https://github.com/macports/macports-ports/pull/24856

**Comment by barracuda156** [comment]
> @sikmir @omar-polo I rebuilt the package on 10.6 machine with OpenSSL installed and it built successfully, but now had a strange effect when running. There's no TUI and only the letter 's' appears (same on iTerm 2):
> 
> image
> 
> Nothing suspicious in otool:
> 
> ```
> otool -L $(which telescope)
> /opt/local/bin/telescope:
>         /opt/local/lib/libMacportsLegacySupport.dylib (compatibility version 1.0.0, current version 1.2.1)
>         /opt/local/libexec/openssl3/lib/libssl.3.dylib (compatibility version 3.0.0, current version 3.0.0)
>         /opt/local/libexec/openssl3/lib/libcrypto.3.dylib (compatibility version 3.0.0, current version 3.0.0)
>         /opt/local/lib/libtls.28.dylib (compatibility version 29.0.0, current version 29.0.0)
>         /opt/local/lib/libncurses.6.dylib (compatibility version 6.0.0, current version 6.0.0)
>         /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 125.2.11)
> ```
> 
> This is starting to remind me of the story in #19047, but `legacysupport` uses the [correct](https://github.com/macports/macports-legacy-support/commit/d24f1d50c27f86fa05aa0fac209b6468b07f4940) `open_memstream` implementation.

@aeiouaeiouaeiouaeiouaeiouaeiou Could you say if 0.10.1 still has the bug or not on your system?

**Comment by aeiouaeiouaeiouaeiouaeiouaeiou** [comment]
@barracuda156 I feel exhausted today for the tests, but yes, the app is still crashing on SL.

**Comment by sikmir** [comment]
```
$ otool -L /opt/local/bin/telescope
/opt/local/bin/telescope:
	/opt/local/libexec/openssl3/lib/libcrypto.3.dylib (compatibility version 3.0.0, current version 3.0.0)
	/opt/local/lib/libtls.28.dylib (compatibility version 29.0.0, current version 29.0.0)
	/opt/local/lib/libncurses.6.dylib (compatibility version 6.0.0, current version 6.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
$ /opt/local/bin/telescope gemini://gemini.omarpolo.com
<... capsula opened, no crash ...>
```

**Comment by barracuda156** [comment]
@sikmir Just to verify, I tried to replicate with the same command which you used.

`telescope` fails:
telescope_gemini

While `offpunk` works:
offpunk_gemini

**Pull request closed without merging**